applications/rogue/main.tf

```diff
@@ -182,17 +184,16 @@ module "queue" {
   user = module.iam_user.name
 }
 
-module "storage" {
-  source = "../../components/s3_bucket"
-
-  application = var.name
-  name        = var.name
-  environment = var.environment
-  stack       = local.stack
+data "aws_s3_bucket" "bucket" {
+  bucket = var.name
+}
 
+resource "aws_iam_user_policy" "s3_policy" {
+  name = "${var.name}-s3"
   user = module.iam_user.name
-
-  replication_target = var.backup_storage_bucket
+  policy = templatefile("${path.module}/s3-iam-policy.json.tpl", {
+    bucket_arn = data.aws_s3_bucket.bucket.arn
+  })
```

Comment:
I replaced the [`module.s3_bucket`](https://github.com/DoSomething/infrastructure/tree/main/components/s3_bucket) module here with a [`s3_bucket` data resource](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/data-sources/s3_bucket) (which _reads_ from an existing bucket by name, rather than creating/modifying it).

This allows us to continue to reference the bucket that we've moved into Northstar's application module. I then created a `aws_iam_user_policy.s3_policy` that allows Rogue's IAM user to continue to access this bucket (since the relocated module will now default to granting access to Northstar).

Reply:
Using a data resource for something that's defined elsewhere is probably a bad practice (as opposed to defining the bucket elsewhere and passing it in as a variable to each app that shares it), but I think this the cleanest way to handle this during this "transition period".